api/docs/Doxyfile

```diff
@@ -839,8 +839,10 @@ RECURSIVE              = YES
 
 EXCLUDE                = ../include/opentelemetry/common/spin_lock_mutex.h \
                          ../include/opentelemetry/common/key_value_iterable_view.h \
+                         ../include/opentelemetry/common/kv_properties.h \
```

Comment:
Could we move these files into a separate folder like `internal` to make it clear that they are not intended for end-user?